addons/hr/models/hr_employee.py

```diff
@@ -187,6 +188,30 @@ def _compute_work_permit_name(self):
             permit_no = '_' + employee.permit_no if employee.permit_no else ''
             employee.work_permit_name = "%swork_permit%s" % (name, permit_no)
 
+    @api.depends('user_id')
```

Comment:
Since we need to add a dependency in `hr_recruitment`, we can add a method to get the dependencies
```suggestion
    def _get_partner_count_depends(self):
        return ['user_id']

    @api.depends(lambda self: self._get_partner_count_depends())
```